app/src/org/commcare/android/framework/CommCareActivity.java

```diff
@@ -238,24 +228,6 @@ protected void loadFields(boolean restoreOldFields) {
                         View v = this.findViewById(element.value());
                         f.set(this, v);
 
-                        if (oldActivity != null && restoreOldFields) {
-                            View oldView = (View) f.get(oldActivity);
-                            if (oldView != null) {
-                                if (v instanceof TextView) {
-                                    ((TextView) v).setText(((TextView) oldView).getText());
-                                }
-                                if (v == null) {
-                                    Log.d("loadFields", "NullPointerException when trying to find view with id: " +
-                                            +element.value() + " (" + getResources().getResourceEntryName(element.value()) + ") "
-                                            + " oldView is " + oldView + " for activity " + oldActivity +
-                                            ", element is: " + f + " (" + f.getName() + ")");
-                                }
-                                v.setVisibility(oldView.getVisibility());
-                                v.setEnabled(oldView.isEnabled());
-                                continue;
-                            }

```

Comment:
This restores the state of fields from the previous state. IE: If you rotate, this ensures that if you've typed text into a field, changed it from visible to GONE, or disabled it, that it will be in the same state in the rotated view.